server/test/test_analytics_endpoint.py

```
@@ -0,0 +1,139 @@
+        import {{ computeLearningAnalytics }} from './dist/services/analyticsService.js';
+
+        const db = JSON.parse(fs.readFileSync({json.dumps(str(db_path))}, 'utf8'));
+        const originalNow = Date.now;
+        Date.now = () => {last_calculated};
+        const result = computeLearningAnalytics(db);
+        Date.now = originalNow;
+        console.log(JSON.stringify(result));
+        """
+    )
+
+    output = subprocess.check_output(
```

Ruff (0.13.3)

`subprocess` call: check for execution of untrusted input

(S603)